src/passes.rs

```diff
@@ -362,6 +372,11 @@ impl<T: PassManagerSubType> PassManager<T> {
     /// makes arguments dead, but does not remove them. The existing
     /// dead argument elimination pass should be run after this to
     /// clean up the mess.
+    ///
+    /// In LLVM 12 and later, this instruction is replaced by the
+    /// [`add_instruction_simplify_pass`].
+    #[llvm_versions(3.6..=11.0)]
+    #[doc(alias = "add_instruction_simplify_pass")]
```

Comment:
doc aliases ensure that when searching for the old (now-deprectated) pass you also find the new one, and vice versa.

Reply:
This is cool, I didn't know about doc alias. But I'm not sure it matters or will even work - this function will be compiled out of llvm 12 builds (which will include our publicly generated/hosted docs)

Reply:
I verified locally that when building with `--features="llvm12-0"` when I search for `add_constant_propagation_pass` I see the alias for the new version. It has to be an exact match though, even `add_constant_propagation_pas` will not find it. But maybe this will improve further in the future.